runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/net/IpAddr.kt

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.net
+
+import aws.smithy.kotlin.runtime.util.InternalApi
+
+/**
+ * An IP Address (either IPv4 or IPv6)
+ */
+@InternalApi
+public sealed class IpAddr {
```

Comment:
**bikeshed**

The variants of this sealed class are currently nested classes (e.g. `IpAddr.Ipv4`). Alternatively we could move each subclass to it's own file (I tend to like scoped names as it usually helps readability/understanding but these are a little bit more involved).

Reply:
> If we stay with this structure

I'm not sure I follow what you're saying or implying with this comment.